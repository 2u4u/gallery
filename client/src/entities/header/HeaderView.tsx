import { View } from 'shared/types/index.type';

type Props = {
    view: View,
    changeView: (view: View) => void;
}

function HeaderView({ changeView, view}: Props) {
    return (
        <div className='items-view'>
            <button type='button' className='items-view__item' onClick={() => changeView(View.list)}>
                <img src={view === View.list ? './img/list-active.svg' : './img/list.svg'} alt="list" />
            </button>
            <button type='button' className='items-view__item' onClick={() => changeView(View.grid)}>
                <img src={view === View.grid ? './img/grid-active.svg' : './img/grid.svg'} alt="grid" />
            </button>
        </div>
    );
}

export default HeaderView;
