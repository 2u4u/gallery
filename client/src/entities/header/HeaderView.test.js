import renderer from 'react-test-renderer';
import HeaderView from './HeaderView';
import { View } from 'shared/types/index.type';

describe('Header view component ', () => {
    it('matches snapshot', () => {
        const view = View.grid;
        const changeView = () => false;
        const component = renderer.create(
            <HeaderView 
                view={view}
                changeView={changeView}
            />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
