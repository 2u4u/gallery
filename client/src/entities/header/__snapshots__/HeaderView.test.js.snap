// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header view component  matches snapshot 1`] = `
<div
  className="items-view"
>
  <button
    className="items-view__item"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="list"
      src="./img/list.svg"
    />
  </button>
  <button
    className="items-view__item"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="grid"
      src="./img/grid-active.svg"
    />
  </button>
</div>
`;
